mantidproject/mantid
# Indirect - Fit Interfaces - Update mini-plot correctly upon changing the selected function

Currently, in the Indirect > Data Analysis 'fit' interfaces, the mini preview-plots are not updated when changing function.

When changing to a function that wasn't previously fit, the plot should be cleared. When changing back to the function that was previously fit, the fit and difference plots should be added to the preview.

This PR, provides common code for this functionality across all the fit interfaces.

**To test:**
1. Navigate to Interfaces > Indirect > Data Analysis.
2. Follow the instructions for each of the following interfaces:
    - **ConvFit**
        1. Provide the supplied \*\_red file in the Sample field and the supplied \*\_res file in the Resolution field (both should be found in a ConvFit folder, given in the test files).
        2. Select any function in the 'Fit Type' drop-down menu (One/Two Lorentzian will produce the most reasonable result).
        3. Check the 'Use' checkbox, found under 'Delta Function' in the property table.
        4. Select a spectra range that does not cover the entire sample workspace range, using the 'Spectra Range' spinners.
        5. Click 'Run'.
        6. Navigate through each spectra in the sample workspace, by changing the 'Plot Spectrum' spinner. For each spectra that is encapsulated in the range selected when 'Run' was clicked, ensure the sample and calculated fit are added to the top plot, and the difference is added to the bottom plot, within the interface. For each spectra not in that range, ensure only the sample is added, in the top plot.
        7. Change the selected function in the 'Fit Type' drop-down menu.
        8. Navigate through each spectra in the sample workspace and ensure only the sample plot is added, in the top plot.
        9. Change back to the function used in the fit.
        10. Navigate through each spectra and perform the same checks as in step f.
    - **I(Q, t) Fit**
        1. Provide the supplied \*\_red file in the Input field  (to be found in an IqtFit folder, given in the test files).
        2. Select any function in the 'Fit Type' drop-down menu (One Exponential will produce the most reasonable result).
        3. Select a spectra range that does not cover the entire sample workspace range, using the 'Spectra Range' spinners.
        4. Click 'Run'.
        5. Navigate through each spectra in the sample workspace, by changing the 'Plot Spectrum' spinner. For each spectra that is encapsulated in the range selected when 'Run' was clicked, ensure the sample and calculated fit are added to the top plot, and the difference is added to the bottom plot, within the interface. For each spectra not in that range, ensure only the sample is added, in the top plot.
        6. Change the selected function in the 'Fit Type' drop-down menu.
        7. Navigate through each spectra in the sample workspace and ensure only the sample plot is added, in the top plot.
        8. Change back to the function used in the fit.
        9. Navigate through each spectra and perform the same checks as in step e.
    - **MSD Fit**
        1. Provide the supplied \*\_red file in the Input field  (to be found in an MSDFit folder, given in the test files).
        2. Select any model in the 'Model' drop-down menu.
        3. Select a spectra range that does not cover the entire sample workspace range, using the 'Spectra Range' spinners.
        4. Click 'Run'.
        5. Navigate through each spectra in the sample workspace, by changing the 'Plot Spectrum' spinner. For each spectra that is encapsulated in the range selected when 'Run' was clicked, ensure the sample and calculated fit are added to the top plot, and the difference is added to the bottom plot, within the interface. For each spectra not in that range, ensure only the sample is added, in the top plot.
        6. Change the selected model in the 'Model' drop-down menu.
        7. Navigate through each spectra in the sample workspace and ensure only the sample plot is added, in the top plot.
        8. Change back to the model used in the fit.
        9. Navigate through each spectra and perform the same checks as in step e.
    - **JumpFit**
        1. Provide the supplied \*\_red file in the Sample field  (to be found in a JumpFit folder, given in the test files).
        2. Select any function in the 'Fit Function' drop-down menu.
        3. Click 'Run'.
        4. Ensure the sample and calculated fit are added to the top plot and the difference is added to the bottom plot.
        6. Change the selected function in the 'Fit-Function' drop-down menu.
        7. Ensure only the sample plot is added, in the top plot.
        8. Change back to the function used in the fit.
        9. Perform the same checks as in step d.
Fixes #21123. 

**Release Notes** 
#### Changes have been added to the release notes.
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
